### PR TITLE
Upgrade chat assistant and chat screen

### LINKED ISSUE #393: Fix the prompt of the AI assistant
-Fix minnor bugs as well that you discovered last sprint

### PR DESCRIPTION
**Pull request by manunam2002**
### Description

This PR introduces new features to the chat AI assistant I implemented last week to make it complete, as well as a new feature in the chat conversation:
- **Translation**: The Assistant Dialog now provides a translation mode in which the user can get his message translated to another language.
- **Message Suggestions**: The Chat Assistant can now analyse the ongoing conversation and provide a list of suggested types of a response.
- **Service Request Access**: A service request is now linked to the conversation when it is initialised, the user can now see a summary of the relevant request in the chat Screen, and the Chat Assistant can use the request to have more context.

### Related Issues

Fixes #393, #394

### Changes Made
- `ChatAssistantDialog`: Introduced the UI for translation mode, accessible via a Switch in the Dialog
- `ChatAssistantViewModel`: Created a `suggestionModel` that outputs a list of suggestions for a response and added the functions to interact with it, modified the previous prompts to make the model more robust in the case of a nonsense conversation.
- `ChatViewModel` , `ChatRepository` , `ServiceRequestViewModel` ,  `ServiceRequestRepository`: Created the necessary functions to implement the link from a chat room to a service request
- `ChatScreen`: Created `RequestDetail` Composable to display the request summary, added `AssistantSuggestions` composable on top of the message input field.
- **Testing**: Updated the existing test to match the new behaviour and added new tests for the testable code

### CODE REVIEW

File: app/src/main/java/com/android/solvit/shared/model/chat/ChatAssistantViewModel.kt [hunk 1]
```diff
@@ -147,16 +189,85 @@ class ChatAssistantViewModel() : ViewModel() {
...
+  /**
+   * Build the prompt for the suggestions assistant
+   *
+   * @return The generated prompt
+   */
+  fun buildSuggestionsPrompt(): String {
+    var prompt = "Provide a list without repetitions of suggestions themes for a response"
+    if (_messageContext.value.isNotEmpty()) {
+      prompt += ", based on the following conversation:\n"
+      prompt += messageContext.value.joinToString("\n") { it.senderName + ": " + it.message }
```
Comment (hjamaly) [inline app/src/main/java/com/android/solvit/shared/model/chat/ChatAssistantViewModel.kt:214]: Method will have to be updated because we use know also image messages but it's normal since the PR handling different format types is not yet merged

File: app/src/test/java/com/android/solvit/shared/model/chat/ChatRepositoryTest.kt
Comment (hjamaly) [inline app/src/test/java/com/android/solvit/shared/model/chat/ChatRepositoryTest.kt:1]: 👍

File: app/src/main/java/com/android/solvit/shared/model/authentication/AuthViewModel.kt
Comment (rimabk) [inline app/src/main/java/com/android/solvit/shared/model/authentication/AuthViewModel.kt:1]: **Issues:**
**Non-null Assertion** (user.value!!): Using !! can lead to crashes if user.value is null, which could happen under certain edge cases (e.g., before the user is fully loaded). Consider adding a null check.

File: app/src/main/java/com/android/solvit/shared/model/authentication/AuthRepository.kt
Comment (rimabk) [inline app/src/main/java/com/android/solvit/shared/model/authentication/AuthRepository.kt:1]: The error handling is clear,but consider adding documentation to the code to enhance maintainability and readability.

File: app/src/main/java/com/android/solvit/shared/model/chat/ChatAssistantViewModel.kt
Comment (rimabk) [inline app/src/main/java/com/android/solvit/shared/model/chat/ChatAssistantViewModel.kt:1]: The `buildSuggestionsPrompt()` function is well-structured. 
Good attention to detail in avoiding repetitions and ensuring suggestions are relevant to the context.
The configuration for the `GenerativeModel` is well thought out for producing concise, relevant responses without overwhelming the user with too much information.

**Review by hjamaly — APPROVED** [review]
**Overview**

This PR introduces key enhancements to the Chat AI Assistant:

- Translation Mode: Translate messages via a switch in the Assistant Dialog.
- Message Suggestions: AI-driven response suggestions based on conversation context.
- Service Request Access: Link service requests to chat, displaying a summary and enhancing context.

This is an impressive effort, and the features appear well-implemented with clear code and effective UI integration. Great work! 🚀

**Strengths**

- Comprehensive Features: Translation, suggestions, and service request integration significantly enhance the chat experience.
- Clean Code: Clear naming and comments make the implementation easy to follow.

**Areas for Improvement**
Testing: Add unit tests for the new ServiceRequestRepository methods to improve coverage

File: app/src/main/java/com/android/solvit/shared/ui/chat/ChatAssistantDialog.kt
Comment (rimabk) [inline app/src/main/java/com/android/solvit/shared/ui/chat/ChatAssistantDialog.kt:1]: The code is well-structured and modular, making it easy to maintain and extend.

File: app/src/main/java/com/android/solvit/shared/ui/chat/ChatScreen.kt
Comment (rimabk) [inline app/src/main/java/com/android/solvit/shared/ui/chat/ChatScreen.kt:1]: This UI is well-organized and modular, with clearly defined composables for each functionality.
Some areas could benefit from better visual feedback or loading indicators, such as when fetching assistant suggestions or sending messages.

**Review by rimabk — APPROVED** [review]
**PR REVIEW:**
This PR successfully introduces well-implemented features that enhance the chat AI assistant, including translation mode, message suggestions, and service request integration. The UI is clean and intuitive, offering a seamless experience for users to interact with the assistant and view relevant service request details. The functionality for generating suggestions based on conversation context is robust and adds significant value to the user interaction. Additionally, the linking of service requests to the conversation is well-executed, improving the overall context for the assistant's responses.

Your efforts in updating the `ChatAssistantViewModel`, `ChatRepository`, and other related components are commendable, ensuring the system works cohesively. The UI changes in `ChatScreen`, including the `RequestDetail` and `AssistantSuggestions `composables, are thoughtfully designed and enhance usability. The added tests ensure the new behavior is well-covered.

Great work overall! You can find my comments in each file for further details.

**Review by aoutz — APPROVED** [review]
This PR adds valuable enhancements to the chat AI assistant, significantly improving its functionality and user experience. The addition of translation mode in the ChatAssistantDialog provides greater versatility, while the message suggestions feature offers users intelligent, context-aware guidance during conversations. Linking service requests to chat rooms is a well-thought-out improvement, enabling users to seamlessly view relevant details and provide better context to the assistant.

The changes to the ChatAssistantViewModel and other components are well-aligned with the new features, and updating prompts to handle edge cases is a great step toward improving model robustness. The inclusion of new tests ensures these additions are reliable and maintainable.

It would be even better if additional documentation and corresponding Figma designs were included to provide a clearer understanding of the feature implementation and alignment with the design. Great work Manuu ! 🚀

File: app/src/main/java/com/android/solvit/shared/model/chat/ChatAssistantViewModel.kt [hunk 2]
```diff
@@ -113,10 +153,10 @@ class ChatAssistantViewModel() : ViewModel() {
    * @param input Additional infos to add to the prompt
    * @return The generated prompt
    */
-  fun buildPrompt(input: String): String {
+  fun buildMessagePrompt(input: String): String {
```
[the remark this answers is not in the document]
Comment (manunam2002) [reply app/src/main/java/com/android/solvit/shared/model/chat/ChatAssistantViewModel.kt:156]: Great suggestion but do you really think it matters? we can discuss it tomorrow

File: app/src/main/java/com/android/solvit/shared/model/chat/ChatRepository.kt [hunk 3]
```diff
@@ -8,6 +8,15 @@ interface ChatRepository {
       receiverUid: String
   )
 
+  fun linkChatToRequest(
+      chatRoomId: String,
+      serviceRequestId: String,
```
[the remark this answers is not in the document]
Comment (manunam2002) [reply app/src/main/java/com/android/solvit/shared/model/chat/ChatRepository.kt:13]: The purpose here is just creating a firestore document that maps a chatroom id to a service request id, it would be inefficient to pass the whole request as a parameter
